src/messaging/ReliableMessageContext.h

```diff
@@ -67,11 +67,10 @@ class ReliableMessageContext
     }
 
     /**
-     * Check whether we have an ack to piggyback on the message we are sending.
-     * If true, TakePendingPeerAckMessageCounter will return a valid value that
-     * should be included as an ack in the message.
+     * Check whether we have a valid message counter. The counter is valid once received a message with a request ack flag. It never
+     * flips to false.
      */
-    bool HasPiggybackAckPending() const;
+    bool HasAckCounter() const;
```

Comment:
Why this change?  The old name accurately described the semantics of the function: if it returns true, there is a piggyback ack to be sent whenever we are sending a message.